fix: ensure that we capture service worker requests (#28517)
cypress-io/cypress: cli/CHANGELOG.md

**Bugfixes:**

> - Fixed asset capture for Test Replay (which will potentially affect styles) specifically for requests that are routed through service workers. Fixes [#28516](https://github.com/cypress-io/cypress/issues/28516).

Reviewer: > which will potentially affect styles

what does this mean/entail? what should on expect by from this?

Author: Resolved here: [`04617eb` (#28517)](https://github.com/cypress-io/cypress/pull/28517/commits/04617eb1b3565c380ac89d12f575f2e731cba4f6)

> - Fixed asset capture for Test Replay for requests that are routed through service workers and for requests that span tests when test isolation is off. This addresses an issue where styles would not being applied properly in Test Replay or `cy.intercept` would not work properly. Fixes [#28516](https://github.com/cypress-io/cypress/issues/28516) and [#28545](https://github.com/cypress-io/cypress/issues/28545).

Reviewer 2: ```suggestion
- Fixed asset capture for Test Replay for requests that are routed through service workers and for requests that span tests when test isolation is off. This addresses an issue where styles were not being applied properly in Test Replay and `cy.intercept` was not working properly for requests in this scenario. Fixes [#28516](https://github.com/cypress-io/cypress/issues/28516) and [#28545](https://github.com/cypress-io/cypress/issues/28545).
```

Not sure if this is a complete correction of the second sentence. Please cross-check!

Author: Thanks @MikeMcC399. Updated.

**Bugfixes:**

- Fixed asset capture for Test Replay for requests that are routed through service workers and for requests that span tests when test isolation is off. This addresses an issue where styles were not being applied properly in Test Replay and `cy.intercept` was not working properly for requests in this scenario. Fixes [#28516](https://github.com/cypress-io/cypress/issues/28516) and [#28545](https://github.com/cypress-io/cypress/issues/28545).

> - Cypress no longer errors during `cypress run` or `cypress open` when using Typescript 5.3.2+ with `extends` in `tsconfig.json`. Fixes [#28385](https://github.com/cypress-io/cypress/issues/28385).

Author: ```suggestion
```
